pkg/datapath/linux/node.go

```diff
@@ -295,15 +295,18 @@ func (n *linuxNodeHandler) deleteDirectRoute(CIDR *cidr.CIDR, nodeIP net.IP) {
 	}
 }
 
-// createNodeRoute creates a route that points the specified prefix to the host
-// device via the router IP
+// createNodeRouteSpec creates a route spec that points the specified prefix to the host
+// device via the router IP. The route is configured with a computed MTU for non-local nodes.
 //
 // Example:
 // 10.10.0.0/24 via 10.10.0.1 dev cilium_host src 10.10.0.1
 // f00d::a0a:0:0:0/112 via f00d::a0a:0:0:1 dev cilium_host src fd04::11 metric 1024 pref medium
 //
-func (n *linuxNodeHandler) createNodeRoute(prefix *cidr.CIDR) (route.Route, error) {
-	var local, nexthop net.IP
+func (n *linuxNodeHandler) createNodeRouteSpec(prefix *cidr.CIDR, isLocalNode bool) (route.Route, error) {
+	var (
+		local, nexthop net.IP
+		mtu            int
+	)
```

Comment:
Is the `var` declare really needed, unless the `ZERO` are intended to be used ? Else we can just do literal assignment

Reply:
In this case the intent is specifically to zero the field if the route is specifically for the local pod CIDR. There's a couple of potential ways to represent this in Go, but the code was already declaring vars at the top anyway so the current approach seems reasonable to me.

Reply:
It's conditionally set at L340 below.